src/shogun/machine/gp/SingleLaplacianInferenceMethod.cpp

```diff
@@ -177,19 +207,18 @@ void CSingleLaplacianInferenceMethod::update_chol()
 
 	if (eigen_W.minCoeff() < 0)
 	{
-		// compute inverse of diagonal noise: iW = 1/W
-		VectorXd eigen_iW = (VectorXd::Ones(m_W.vlen)).cwiseQuotient(eigen_W);
-
+		//A = eye(n)+K.*repmat(w',n,1);
 		FullPivLU<MatrixXd> lu(
-			eigen_ktrtr*CMath::sq(m_scale)+MatrixXd(eigen_iW.asDiagonal()));
-
-		// compute cholesky: L = -(K + iW)^-1
-		eigen_L = -lu.inverse();
+			MatrixXd::Identity(m_ktrtr.num_rows,m_ktrtr.num_cols)+
+			eigen_ktrtr*CMath::sq(m_scale)*eigen_W.asDiagonal());
+		// compute cholesky: L = -(K + 1/W)^-1
+		//-iA = -inv(A)
+		eigen_L=-lu.inverse();
```

Comment:
@lambday sorry for constantly pinging here, but how are the matrix factorisations doing? :)